R/get_future_workshops.R

```diff
@@ -1,40 +1,90 @@
-           instructor = paste(lead_instructor, supporting_instructor1, supporting_instructor2, sep = ", "),
-           instructor = gsub(", NA", "", instructor),
-           helper = paste(helper1, helper2, helper3, sep=", "),
-           helper = gsub(", NA", "", helper),
-           latitude = ifelse(address=="online",NA, osm_search(address, key=token)$lat), #TODO: pass Open street maps token...
-           longitude = ifelse(address=="online",NA, osm_search(address, key=token)$lon) #... as a variable
-    )
-  return(dat_struct)
+get_future_workshops <- function(df, future = "today") {
+  #TODO verify input for future parameter
+
+  # verify that data contains the correct columns
+  columns_needed <- c("startdate", "enddate", "starttime", "endtime",
+                      "title", "slug",
+                      "lead_instructor", "supporting_instructor1", "supporting_instructor2",
+                      "helper1", "helper2", "helper3",
+                      "carpentry", "curriculum", "flavor", "host",
+                      "venue", "address", "country",
+                      "eventbrite", "repository", "ready")
+
+  for(c in columns_needed){
+    if(!c %in% names(df)){
+      e <- paste("Column", c, "is not present in the data.\
+                 Please verify whether the data is complete, and column names are correct.")
+      stop(e)
+    }
+  }
+
+  df <- tidyr::drop_na(df, title)
+  df <- df[columns_needed]
+
+  # define dates and times in human readable formats
+  df <- dplyr::mutate(df,
+                      startdate = convert_to_date(startdate),
+                      enddate = convert_to_date(enddate),
+                      humandate = human_date(startdate,enddate),
+                      humantime = human_time(starttime,endtime,enddate)
+  )
+
+  # only pick future workshops, if requested
+  if(future != "none"){
+    # the following is hacky. I do not know why this is necessary, but it works.
+    if(future == "today"){
+      date <- as.character(lubridate::today(tzone="Europe/Amsterdam"))
+      date <- convert_to_date(date)
+    } else{
+      date <- convert_to_date(future)
+    }
+    df <- dplyr::filter(df, startdate >= date)
+  }
+
+  # format helpers and instructors
+  df <- dplyr::mutate(df,
+                      instructor = list_people(lead_instructor, supporting_instructor1, supporting_instructor2),
+                      helper = list_people(helper1, helper2, helper3)
+  )
+
+  # add latitude/longitude
+  df <- dplyr::mutate(df,
+                        latitude = ifelse(address=="online",NA, nominatimlite::geo_lite(address)$lat),
+                        longitude = ifelse(address=="online",NA, nominatimlite::geo_lite(address)$lon)
+  )
+  return(df)
+}
+
+convert_to_date <- function(date){
+  lubridate::ymd(date, tz="Europe/Amsterdam")
+}
+
+human_date <- function(startdate,enddate){
+  ifelse(months(startdate)==months(enddate), #this includes the month for end date only when workshop go over month switch
```

Comment:
I love that you added this!

Reply:
That was verbatim from the original code ;) just turned into a function.